oxidation/libparsec/crates/crypto/src/sodiumoxide/secret.rs

```diff
@@ -47,8 +47,13 @@ impl SecretKey {
         Ok(plaintext)
     }
 
+    /// # Safety
+    ///
+    /// This function requires access to libsodium methods that are not
+    /// exposed directly, so it uses the unsafe C API
+    /// ...
     pub fn hmac(&self, data: &[u8], digest_size: usize) -> Vec<u8> {
-        // Sodiumoxide doesn't expose those methods, so we have to access
+        // SAFETY: Sodiumoxide doesn't expose those methods, so we have to access
         // the libsodium C API directly
```

Comment:
This doesn't explain why the code bellow is safe

Reply:
I am not sure why/if the code is safe. I will be looking at sodiumoxide's docs because those function might be provided now (git blame informs me this function was written 10 months ago).

I can't tell for sure why or if this function is safe, but I might be able to get rid of the unsafe{} block this way.